### DISCUSSION TITLE
posts/how-to-build-an-image-picker-in-react-native/

### DISCUSSION THREAD

**Discussion by giscus** [yjose/elazizi.com#94] [Blog posts]
# posts/how-to-build-an-image-picker-in-react-native/

In this tutorial, we'll demonstrate how to use React Native Image Crop Picker to enable your users to select images from their devices or use the device's camera to capture and upload live photos to your app.

https://elazizi.com/posts/how-to-build-an-image-picker-in-react-native/

**Comment by mwaleed-39** [comment]
I faced some issue in react native. when we pressed upload image, the error message shows:
CODE:

import React, { useState } from 'react'
import { Alert, Button, View,StyleSheet } from 'react-native'
import * as ImagePicker from 'react-native-image-picker'
import {launchImageLibrary} from 'react-native-image-picker'

const options ={
title: 'Select Image',
type: 'library',
options: {
maxHeight:200,
maxWidth:200,
selectionLimit: 1,
mediaType: 'photo',
includeBase64: false,
}
}

const Request = () => {
const opengallery = async()=>{
const image = await ImagePicker.launchImageLibrary(options);
console.log(image)
}

return (



)
}
export default Request;

const styles = StyleSheet.create({
container:{
flex:1,
alignItems:'center',
justifyContent:'center'
}
})

when we pressed upload button, the error message shows:

Possible Unhandled Promise Rejection (id: 1):
TypeError: Cannot read property 'launchImageLibrary' of null
TypeError: Cannot read property 'launchImageLibrary' of null

I create API:
app.post('/finerequest', (req, res, next) => {

const file = req.files.image;
cloudinary.uploader.upload(file.tempFilePath, (err, result) => {
console.log(result)
const requestId = req.body.request_id
const fine = new Fine({
_id: new mongoose.Types.ObjectId,
request_id: requestId,
image: result.url
})
fine.save()
.then(result => {
res.status(200).json({
message: "Fine and Vehicle Repair Request added",
response: result
})
})
.catch(err => {
res.status(500).json({
message: "Fine and Vehicle request not added",
error: err
})
})
})
})
How to call this API, when we upload image in front end side, the image will be stored in mongodb
